client/extensions/woocommerce/app/settings/taxes/index.js

```diff
@@ -9,207 +9,92 @@ import PropTypes from 'prop-types';
 import { bindActionCreators } from 'redux';
 import classNames from 'classnames';
 import { connect } from 'react-redux';
+import { find, isEmpty } from 'lodash';
 import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-import ActionHeader from 'woocommerce/components/action-header';
-import {
-	areSettingsGeneralLoaded,
-	areTaxCalculationsEnabled,
-} from 'woocommerce/state/sites/settings/general/selectors';
-import {
-	areTaxSettingsLoaded,
-	getPricesIncludeTax,
-	getShippingIsTaxFree,
-} from 'woocommerce/state/sites/settings/tax/selectors';
-import ExtendedHeader from 'woocommerce/components/extended-header';
-import { updateTaxesEnabledSetting } from 'woocommerce/state/sites/settings/general/actions';
-import QuerySettingsGeneral from 'woocommerce/components/query-settings-general';
-import { fetchTaxRates } from 'woocommerce/state/sites/meta/taxrates/actions';
-import { fetchTaxSettings, updateTaxSettings } from 'woocommerce/state/sites/settings/tax/actions';
-import { getLink } from 'woocommerce/lib/nav-utils';
+import { fetchPlugins } from 'state/plugins/installed/actions';
+import { getPlugins, isRequesting } from 'state/plugins/installed/selectors';
 import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
 import Main from 'components/main';
-import TaxSettingsSaveButton from './save-button';
-import SettingsNavigation from '../navigation';
-import { successNotice, errorNotice } from 'state/notices/actions';
-import StoreAddress from 'woocommerce/components/store-address';
-import TaxesOptions from './taxes-options';
-import TaxesRates from './taxes-rates';
+import SettingsTaxesPlaceholder from './taxes-placeholder';
+import SettingsTaxesTaxJar from './taxes-taxjar';
+import SettingsTaxesWooCommerceServices from './taxes-wcs';
 
 class SettingsTaxes extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			isSaving: false,
-			pricesIncludeTaxes: props.pricesIncludeTaxes,
-			shippingIsTaxable: props.shippingIsTaxable,
-			taxesEnabled: props.taxesEnabled,
-			userBeganEditing: false,
-		};
-	}
-
 	static propTypes = {
-		site: PropTypes.shape( {
-			slug: PropTypes.string,
-			ID: PropTypes.number,
-		} ),
 		className: PropTypes.string,
+		isRequestingSitePlugins: PropTypes.bool,
+		siteId: PropTypes.number,
+		sitePluginsLoaded: PropTypes.bool,
+		siteSlug: PropTypes.string,
+		taxJarPluginActive: PropTypes.bool,
 	};
 
 	componentDidMount = () => {
-		const { site } = this.props;
+		const { isRequestingSitePlugins, siteId, sitePluginsLoaded } = this.props;
 
-		if ( site && site.ID ) {
-			this.props.fetchTaxSettings( site.ID );
+		if ( siteId && ! isRequestingSitePlugins && ! sitePluginsLoaded ) {
```

Comment:
The only thing that I thought could go awry here is that potentially site plugins could become stale in the context of this component. Since `sitePluginsLoaded` just checks to see if the plugins array is empty, we could potentially not fetch the latest plugin data here... i.e. I change plugin status outside of calypso.

Reply:
If you are on the Settings > Taxes page, then go to Plugins and install and activate TaxJar, then return to the Settings > Taxes page, you'll have this problem too. I'll come up with a fix.

Reply:
It isn't super critical imho, but figured it is worth noting. Could be handled in a subsequent PR.

Reply:
OK, in https://github.com/Automattic/wp-calypso/pull/19535/commits/2e6ebd33b7e8c239042ba2e773448cabf754c007 I made a change so that if you install/activate TaxJar from Calypso and return to the Taxes Settings view, at least we'll update appropriately.  It doesn't solve the problem of navigating to wp-admin and messing there, but it is a start.

I've written https://github.com/Automattic/wp-calypso/issues/19928 for the case where they use another window/tab to deactivate/activate the TaxJar plugin.